packages/codegen-ui-react/lib/forms/component-helper.ts

```diff
@@ -0,0 +1,67 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import { FieldConfigMetadata } from '@aws-amplify/codegen-ui';
+import { factory, SyntaxKind } from 'typescript';
+
+export const ControlledComponents = ['StepperField', 'SliderField', 'SelectField'];
```

Comment:
SwitchField and ToggleButton are most likely controlled as well. What's the impact of not including all controlled components?

Reply:
the impact is minimal though the idea is to not render default values for these controlled components as the value is coming from the state - will add the other two